src/ukify/ukify.py

```diff
@@ -64,6 +64,24 @@ def shell_join(cmd):
     return ' '.join(shlex.quote(str(x)) for x in cmd)
 
 
+def check_path_exists(p):
+    """Immediately check whether the file exists and is readable.
+
+    In particular initrds may not be readable by non-root users, so we open and
+    close the file. This is quick, but will throw a nice exception for us if
+    access is not possible.
+    """
+    if p is not None:
+        p.open().close()
+
+
+def path_must_exist(s):
+    """Convert a filename string to a Path, checking for existence."""
+    p = pathlib.Path(s)
+    check_path_exists(p)
+    return p
```

Comment:
The functionality between `check_path_exists` and `path_must_exist` seems a bit doubled, what about
```suggestion
def path_is_readable(s: str | None) -> pathlib.Path | None:
    """Convert a filename string to a Path, checking for existence."""
    if s is None:
        return
    p = pathlib.Path(s)
    p.open().close()
    return p
```